Assets/Scripts/Battlefield/TimescaleManager.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TimescaleManager : MonoBehaviour
+{
+   public void SpeedUpTime()
+    {
+        Time.timeScale = 4;
+    }
+    public void SlowDownTime()
+    {
+        Time.timeScale = .25f;
+    }
+    public void SetNormalTime()
+    {
+        Time.timeScale = 1;
+    }
```

Comment:
This would be better to set through a [serializable] private value and the methods make more sense to name them by the timescale set to since SlowDownTime sound like it should always decrease time but it doesn't (i.e SetTimeOneFourth() and SetTimeFour()). Also this doesn't need to be and probably should not be a MonoBehaviour but it should be a normal class that is an object in TurnManager since TurnManager should "manage the time".

Reply:
I named the methods generically so that we can mess around with the values of time without locking in on a specific time value for speeding up and slowing down. And I'm not sure I'm seeing why this being attached to the turn manager is necessary or better because it is not related to the turn manager at all. I feel like this way is less coupled and because the turn manager has nothing to do with the buttons I added I don't think it should be a child of it

Reply:
And the methods have to be public or they can't be called by the buttons (from what youtube says)

Reply:
Forget about what I said about TurnManager. I forgot that this speed is controlled by the GUI buttons. But I still dont believe that it needs to be a MonoBehavior right?

Reply:
I watched a few tutorials for adding functional GUI buttons and all of them derived from monobehavior so I'm not sure what it would be if it wasn't

Reply:
@subashch6 for the names of the functions, we can set those to be more specific once we all agree on what values we want them to adjust the timescale to

Reply:
I mean, it could just be a regular class right? One that the Canvas can interact with. Like a Singleton and not a MonoBehavior?

Reply:
TBH I don't really know enough about unity to answer that, but is there any down side to it being monobehavior?

Reply:
@subashch6 I just tried to make it not a monobehavior but I got tons of errors and the buttons didn't work so i think it has to be one

Reply:
Alright then, I think it might have just been my ignorance of the unity system. I thought it would work. I will accept this pull request. My Bad!